bin/rmt-cli

```diff
@@ -1,29 +1,27 @@
 #!/usr/bin/env ruby
 
-require 'etc'
-
 rmt_path = File.expand_path('..', __dir__)
 require_relative '../config/boot'
 $LOAD_PATH.unshift File.join(rmt_path, 'lib')
 
+require 'etc'
 require 'active_support'
 require 'active_record'
 require 'erb'
 require 'yaml'
 require 'rmt/config'
 
-# don't run as root
-if Process.euid == 0
+relative_load_paths = %w[lib lib/rmt app/models app/services].map { |dir| File.join(rmt_path, dir) }
+ActiveSupport::Dependencies.autoload_paths += relative_load_paths
+
+if RMT::CLI::Base.process_user_name == 'root'
   # set group and then user, otherwise user cannot change group
   Process::Sys.setegid(Etc.getgrnam(RMT::DEFAULT_GROUP).gid)
   Process::Sys.seteuid(Etc.getpwnam(RMT::DEFAULT_USER).uid)
 end
 
-relative_load_paths = %w[lib lib/rmt app/models app/services].map { |dir| File.join(rmt_path, dir) }
-ActiveSupport::Dependencies.autoload_paths += relative_load_paths
-
 if File.exist?(RMT::DEFAULT_MIRROR_DIR) && !File.writable?(RMT::DEFAULT_MIRROR_DIR)
```

Comment:
I still think this can be unified with `needs_path`, but otherwise LGTM.

Reply:
Well, in theory I agree, in practice this doesn't work out so clean:
(1) Since `./bin/rmt-cli` is not inheriting from `RMT::CLI:base`, `needs_path` isn't available here, as it is not inside of `class << self` of `RMT::CLI::Base`. We could change this, but than we need to prefix every `needs_path` with `RMT::CLI::Base`
(2) Even if we did this, we still wouldn't get the nice error handling, since the Exceptions `needs_path` throws, are not caught by the Thor class

Reply:
(1) (2) Override `RMT::CLI::Main.start` and move into that method. `RMT::CLI::Main` inherits from `RMT::CLI::Base` and handles exceptions in a nice way.

Reply:
Indeed. But it was a fun thing to try. :grimacing::+1: